regression_tests/tests/gitbook/test_docs.py

```diff
@@ -0,0 +1,164 @@
+    "workflows/managing-workflow-schedules.md": [
+# Skipping because references workflows by UUID.
+"""workflow_a = client.flow('8fb25dc4-62ed-44a3-872d-c3ff988c8dd3')
+
+# source_flow can be a Flow object, workflow name, or workflow ID
+flow = client.publish_flow(name='workflow_b', 
+                           artifacts=[data],
+                           source_flow=source_flow)""",
+# Skipping because references workflows by UUID.
+"""workflow_id = "0c007eff-6ae0-4a1a-a114-5f16164ffcdf" # Set your workflow ID here.
+client.trigger(id=workflow_id)""",
+    ],
+}
+def get_code(page):
+    contents = Path(page).read_text()
+    return re.findall("`{3}python\n([\s\S]*?)\n`{3}", contents)
+
+def run_in_order(snippet_list, filename):
+    success = True
+    error = ""
+    output = ""
+    i = len(snippet_list)
+
+    try:
+        # Create a temporary file
+        with open(filename, "w") as f:
+            f.write("\n".join(snippet_list))
+
+        # Execute the code in the file and capture the output
+        result = subprocess.run(["python3", filename], stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True, text=True)
+
+        output = result.stdout
+    except subprocess.CalledProcessError as e:
+        if i <= 1:
+            success = False
+            output = e.stdout
+            error = e.stderr
+        else:
+            # Determine which block is the issue.
+            for i in range(1, i):
+                try:
+                    with open(filename, "w") as f:
+                        f.write("\n".join(snippet_list[:i]))
+                    result = subprocess.run(["python3", filename], stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True, text=True)
+                except subprocess.CalledProcessError as e:
+                    success = False
+                    output = e.stdout
+                    error = e.stderr
+                    i -= 1
+                    break
+    # Delete the temporary file
+    os.remove(filename)
+    return SnippetExec(snippet_list, success, output, i, error)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--docs-folder",
+        dest="docs_folder",
+        default="gitbook",
+        action="store",
+        help="Path to gitbook directory.",
+    )
+
+    parser.add_argument(
+        "--run-subset",
+        dest="run_subset",
+        default=False,
+        action="store_true",
+        help="Run a subset of the gitbook sections.",
+    )
+
+    parser.add_argument(
+        "--sections",
+        dest="sections",
+        default=[],
+        nargs='+',
+        action="store",
+        help="Subset of gitbook sections to run.",
+    )
+
+    args = parser.parse_args()
+
+    if args.run_subset:
+        args.sections = [(section, 0) for section in args.sections_subset]
+    else:
+        args.sections = os.walk(args.docs_folder)
+
+    snippets_by_page = {}
+
+    successfully_ran_all = True
```

Comment:
Both this and `run_in_order()` are super opaque and I'm having a hard time figuring out what the purpose of each line is. Do you mind refactoring these into more bite-sized chunks and documenting the purpose of each? You don't need to go into the specifics of each line but I would like to understand the purpose of each chunk.